flink-yarn/src/main/java/org/apache/flink/yarn/YarnClusterDescriptor.java

```diff
@@ -1195,9 +1184,22 @@ private ApplicationReport startAppMaster(
         yarnClient.submitApplication(appContext);
 
         LOG.info("Waiting for the cluster to be allocated");
+
+        waitAppSubmission(yarnClient, appId, false);
```

Comment:
Instead of using booleans to encode the behaviour can't we define a target state for which we want to wait? E.g. here I want to wait until we see the `ACCEPTED` state.

Reply:
Reworked.